wandb/apis/public.py

```diff
@@ -2978,11 +2980,12 @@ def from_id(cls, artifact_id, client):
 
             return artifact
 
-    def __init__(self, client, entity, project, name, attrs=None):
+    def __init__(self, client, entity, project, name, attrs=None, use_as=None):
```

Comment:
do we expect this version of `use_as` to be set by user facing code? It _seems_ like this is just used as bookkeeping and passed down through the `use_artifact` call site. If that reading is correct, consider making this private via an underscore, or better yet, a private setter method.

Reply:
I think you're right. I'll do that.